### DISCUSSION TITLE
Create a tigger that contains rich text/formatting

### DISCUSSION THREAD

**Discussion by xixixi721** [espanso/espanso#1302] [Q&A]
Hi all,

I'm trying to create an espanso command that will generate a table for me to easily enter feedback for my students.

https://imgur.com/gallery/wkVDscD

r/espanso - Create a tigger that contains rich text/formatting

Unfortunately, the code below is not working in espanso, resulting in this error.

Caused by:

0: failed to parse YAML match group

1: while parsing a block mapping, did not find expected key at line 47 column 22

2: while parsing a block mapping, did not find expected key at line 47 column 22

The code:

https://pastebin.com/kLWiHvBs

**Comment by krystof-k** [comment]
Hey, you have to use YAML blocks for multiline content. Look [at this](https://yaml-multiline.info).

This should work:

```yaml
  - trigger: "!fb"
    html: |
      <p style='margin-right:0cm;margin-left:0cm;font-size:16px;font-family:"Calibri",sans-serif;margin:0cm;'><strong><span style="font-size:13px;">Comments:</span></strong></p>
      <p style='margin-right:0cm;margin-left:0cm;font-size:16px;font-family:"Calibri",sans-serif;margin:0cm;'><span style="color:red;">Good effort on this task \_\_\_\&nbsp;</span><span style='font-family:"Apple Color Emoji";color:red;'>😊</span><span style="color:red;">\&nbsp;</span></p>
      <p style='margin-right:0cm;margin-left:0cm;font-size:16px;font-family:"Calibri",sans-serif;margin:0cm;'><span style="color:red;">Please review my comments to the right and contact me if you have any questions.</span></p>
      <table style="width:490.1pt;border-collapse:collapse;border:none;">
      <tbody>
      <tr>
      <td style="width: 242.05pt;border: 1pt solid windowtext;background: rgb(146, 208, 80);padding: 0cm 5.4pt;height: 19.5pt;vertical-align: top;">
      <p style='margin-right:0cm;margin-left:0cm;font-size:16px;font-family:"Calibri",sans-serif;margin:0cm;'><span style='font-family:"Calibri",sans-serif;color:black;'>Positive\&nbsp;</span><span style='font-family: "Apple Color Emoji";color:black;'>⭐️</span></p>
      </td>
      <td style="width: 248.05pt;border-top: 1pt solid windowtext;border-right: 1pt solid windowtext;border-bottom: 1pt solid windowtext;border-image: initial;border-left: none;background: rgb(255, 229, 153);padding: 0cm 5.4pt;height: 19.5pt;vertical-align: top;">
      <p style='margin-right:0cm;margin-left:0cm;font-size:16px;font-family:"Calibri",sans-serif;margin:0cm;'><span style='font-family:"Calibri",sans-serif;color:black;'>Improvement\&nbsp;</span><span style='font-family:"Apple Color Emoji";color:black;'>🛠</span></p>
      </td>
      </tr>
      <tr>
      <td style="width: 242.05pt;border-right: 1pt solid windowtext;border-bottom: 1pt solid windowtext;border-left: 1pt solid windowtext;border-image: initial;border-top: none;background: rgb(146, 208, 80);padding: 0cm 5.4pt;height: 88.4pt;vertical-align: top;">
      <div style='margin:0cm;font-size:16px;font-family:"Calibri",sans-serif;'>
      <ul style="margin-bottom:0cm;list-style-type: undefined;margin-left:26px;">
      <li style='margin:0cm;font-size:16px;font-family:"Calibri",sans-serif;'><span style='font-family:"Calibri",sans-serif;color:black;'>Expression is clear</span></li>
      <li style='margin:0cm;font-size:16px;font-family:"Calibri",sans-serif;'><span style='font-family:"Calibri",sans-serif;color:black;'>Good ideas presented in your writing\&nbsp;</span></li>
      <li style='margin:0cm;font-size:16px;font-family:"Calibri",sans-serif;'><span style='font-family:"Calibri",sans-serif;color:black;'>Good structure</span></li>
      </ul>
      </div>
      <p style='margin-right:0cm;margin-left:0cm;font-size:16px;font-family:"Calibri",sans-serif;margin:0cm;'><span style='font-family:"Calibri",sans-serif;'>\&nbsp;</span></p>
      </td>
      <td style="width: 248.05pt;border-top: none;border-left: none;border-bottom: 1pt solid windowtext;border-right: 1pt solid windowtext;background: rgb(255, 229, 153);padding: 0cm 5.4pt;height: 88.4pt;vertical-align: top;">
      <div style='margin:0cm;font-size:16px;font-family:"Calibri",sans-serif;'>
      <ul style="margin-bottom:0cm;list-style-type: undefined;margin-left:26px;">
      <li style='margin:0cm;font-size:16px;font-family:"Calibri",sans-serif;'><span style='font-family:"Calibri",sans-serif;color:black;'>Grammar and spelling need \&nbsp;\&nbsp;</span></li>
      </ul>
      </div>
      </td>
      </tr>
      </tbody>
      </table>
      <p><br></p>
```

**Reply by xixixi721** [reply]
Thank you for your response. Unfortunately, the above code is not working.

When entered it does not produce anything on the page.

**Reply by krystof-k** [reply]
It is working all right on my side.

**Reply by xixixi721** [reply]
Thank you krystof-k :) 

I managed to get this working on my Windows PC after I updated espanso however when I use this command on my Mac it does not show up with the emojis

Any ideas?
![image](https://user-images.githubusercontent.com/109728267/193482954-f854b50a-5de7-4492-aa3d-34061bfd1550.png)

**Reply by eXsoR65** [reply]
@xixixi721 I know this is old, but I had a similar issue so I figure I share what I found. 

For Emojis to work correctly, you need to first make sure that you set the character encoding to UTF-8, otherwise they may not sure up correctly. To so this is simple. Inside your head tag, be sure to specify the following meta tag: `<meta charset="UTF-8">`

Something else you can try is to use Emoji Codepoint instead of your keyboard to place the emojis. 

Here a good article that emplanes all of this: [https://www.kirupa.com/html5/emoji.htm](https://www.kirupa.com/html5/emoji.htm)

**Reply by berot3** [reply]
adding the meta is the only way it works for me too… Still it seems like a bug… why is espanso not using utf-8?

**Reply by smeech** [reply]
Is it Espanso that's the problem?
I added https://espanso.org/docs/matches/extensions/#utf-8-characters to the documentation as a result of other Issues. It looks as though I may need to add an entry for `html:`?

**Reply by krystof-k** [reply]
@smeech you can also add `export LANG='cs_CZ.UTF-8'` for Bash as in [here](https://github.com/espanso/espanso/discussions/1319#discussioncomment-5551587).

**Reply by smeech** [reply]
> @smeech you can also add `export LANG='cs_CZ.UTF-8'` for Bash as in [here](https://github.com/espanso/espanso/discussions/1319#discussioncomment-5551587).

Thank you - I'll add that too.
